src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -72,24 +72,41 @@ public ProductService(BaseDTO dto) {
         return components;
     }
 
+    public Set<Locale> getSupportedLocales() {
+        return getSupportedLocales(true);
+    }
+
     /**
-     * Retrieves the list of locales of a product. It recursively applies data source fallback mechanism in case of failure.
+     * Retrieves the list of locales of a product.
      *
+     * @param withCacheRefresh If true, it recursively applies data source fallback mechanism in case of failure.
      * @return list of locales of the product specified in the dto object, or an empty list in case of failure to retrieve from any data source.
      */
-    public Set<Locale> getSupportedLocales() {
-        return langTagtoLocaleSet(getSupportedLanguageTags());
+    public Set<Locale> getSupportedLocales(boolean withCacheRefresh) {
+        return langTagtoLocaleSet(getSupportedLanguageTags(withCacheRefresh));
+    }
+
+    public Set<Locale> getSupportedLocales(boolean withCacheRefresh, DataSourceEnum dataSource) {
```

Comment:
I think it's not good to bring so many parameters to method and so many similar methods. This makes it difficult to read and hard to maintain.
Please improve the design.

Reply:
What is your suggested design then? Let's be more clear so we could move faster.

Reply:
See the updated code. I have cleaned up some methods.